BE013, BE017

Norshyee opened the pull request:
Zaimplementowano klasy modelowe wraz z adnotacjami JPA

Commit:
build: add entity models

add entity models without hibernate annotations

Norshyee committed:
feat: add model classes

Add model classes as entities, with apropriate hibernate / JPA annotations

Commit:
Merge branch 'develop' into BE013

Commit:
feat: remove liquibase

Remove liquibase from POM and properties

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/actors/Visitor.java:
```diff
@@ -0,0 +1,26 @@
+package pl.envelo.moovelo.entity.actors;
+
+import liquibase.pro.packaged.I;
```
niepotrzebny import liquibase

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/categories/Category.java:
```diff
@@ -0,0 +1,25 @@
+
+@Entity
+@Data
+@AllArgsConstructor
```
Czy w przypadku encji z generowanym id, nie powinno być konstruktora przyjmującego wszystkie wymagane parametry oprócz id? Dotyczy większości klas modelowych, do dyskusji

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/categories/CategoryProposal.java:
```diff
@@ -0,0 +1,28 @@
+
+    private String description;
+
+    private LocalDateTime date;
```
może dodać od razu adnotację do date? 
@DateTimeFormat(pattern = "yyyy-MM-dd")

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/EventInfo.java:
```diff
@@ -0,0 +1,43 @@
+
+    private String description;
+
+    private LocalDateTime startDate;
```
może dodać od razu adnotację do startDate?
@DateTimeFormat(pattern = "yyyy-MM-dd")

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/EventInfo.java:
```diff
@@ -0,0 +1,43 @@
+
+    private LocalDateTime startDate;
+
+    @OneToOne(mappedBy = "eventInfo")
```
Zastanawiam się, czy tu nie powinno być @ManyToOne -> wiele eventów może być w 1 lokalizacji, np. na Cybernetyki 9, więc może nie ma potrzeby za każdym razem tworzyć nowej lokalizacji, tylko wyszukać po id w bazie

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/EventInfo.java:
```diff
@@ -0,0 +1,43 @@
+
+    private Boolean isConfirmationRequired;
+
+    @OneToOne
```
Tutaj bym zmieniła adnotację. Jedna kategoria może być przypisana do wielu EventInfo, myślę, że @ManyToOne od strony Eventu

RyszardEnv replied:
Dokładnie tak powinno być.

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/Event.java:
```diff
@@ -0,0 +1,56 @@
+            joinColumns = @JoinColumn(name = "event_id"),
+            inverseJoinColumns = @JoinColumn(name = "hashtag_id")
+    )
+    private List<Hashtag> hashtags;
```
@ManyToMany(cascade = CascadeType.ALL)

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/groups/Group.java:
```diff
@@ -0,0 +1,42 @@
+    @OneToMany(mappedBy = "group")
+    private List<InternalEvent> events;
+
+    private LocalDateTime creationDate;
```
@DateTimeFormat(pattern = "yyyy-MM-dd")

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/surveys/Answer.java:
```diff
@@ -0,0 +1,28 @@
+    @ManyToOne
+    private EventSurvey eventSurvey;
+
+    private String answerValue;
```
nazwa pola w diagramie klas to 'value', nie 'answerValue'

Norshyee replied:
value się gryzie z H2, tak samo user i group

mdziewiur replied:
ok, zmieniłam nazwę pola w diagramie klas na answerValue

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/surveys/EventSurvey.java:
```diff
@@ -0,0 +1,35 @@
+
+    private String question;
+
+    @OneToMany(mappedBy = "eventSurvey")
```
kaskady, najpierw musimy zapisać odpowiedź, żeby zapisać ankietę

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/Comment.java:
```diff
@@ -0,0 +1,39 @@
+    @ManyToOne
+    private Event event;
+
+    @OneToMany(mappedBy = "comment")
```
CascadeType.All?

RyszardEnv replied:
Warto dorzucić

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/Comment.java:
```diff
@@ -0,0 +1,39 @@
+    private BasicUser basicUser;
+
+    @ManyToMany
+    private List<Hashtag> hashtags;
```
komentarze chyba ostatecznie nie mają listy hashtagów

RyszardEnv replied:
Tak, Hashtag jako Encja funkcjonuje tylko w relacji z Eventem.

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/Comment.java:
```diff
@@ -0,0 +1,39 @@
+
+    private String text;
+
+    private LocalDateTime date;
```
@DateTimeFormat(pattern = "yyyy-MM-dd")

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/Hashtag.java:
```diff
@@ -0,0 +1,38 @@
+            joinColumns = @JoinColumn(name = "hashtag_id"),
+            inverseJoinColumns = @JoinColumn(name = "comment_id")
+    )
+    private List<Comment> comments;
```
nie jestem pewna, czy klasa Hashtag powinna mieć listę komentarzy, ta funkcjonalność była dyskutowana z Kamilem. Do dyskusji z całym backendem

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/Location.java:
```diff
@@ -0,0 +1,35 @@
+    Long id;
+
+    @OneToOne
+    EventInfo eventInfo;
```
@OneToMany

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/actors/User.java:
```diff
@@ -0,0 +1,34 @@
+    private Long id;
+    private String login;
+    private String password;
+    private Role role;
```
@Enumerated(EnumType.STRING)

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/Event.java:
```diff
@@ -0,0 +1,56 @@
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Event {
```
brakuje pola List<BasicUser> usersWithAccess

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/InternalEvent.java:
```diff
@@ -0,0 +1,51 @@
+    private Group group;
+
+    @OneToMany
+    private List<BasicUser> invited;
```
W klasie InternalEvent nie ma pola invited, jest za to w klasie Event ->  List<<BasicUser>> usersWithAccess

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/events/CyclicEvent.java:
```diff
@@ -0,0 +1,46 @@
+        super(group);
+    }
+
+    public CyclicEvent(List<BasicUser> users) {
+        super(users);
+    }
```
czy ten konstruktor jest potrzebny? jeśli przyjmujemy grupę jako parametr konstruktora, to powinniśmy z automatu pobrać listę members z grupy i dodać ją do listy usersWithAccess w klasie Event.

mdziewiur commented on src/main/java/pl/envelo/moovelo/entity/surveys/EventSurvey.java:
```diff
@@ -0,0 +1,35 @@
+    private Long id;
+
+    @OneToMany
+    private List<BasicUser> surveys;
```
nazwa zmiennej surveyees

mdziewiur reviewed (changes requested):
Rafale, czapki z głów!

RyszardEnv commented on src/main/java/pl/envelo/moovelo/entity/Attachment.java:
```diff
@@ -0,0 +1,36 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private File file;
+
```
W bazie danych powinniśmy zapisywać ścieżkę do pliku, a nie plik. Plików nie przetrzymuje się w bazie. Tzn. są takie rozwiązania, ale w naszym przypadku skupiłbym się na ścieżce.

Commit:
fix: PR comments

Add all propositions from PR - https://github.com/akademia-envelo-3/moovelo-back/pull/159
